src/components/SearchMap.js

```diff
@@ -56,20 +65,94 @@ const SearchMap = ({mapLoaded, lastKnownRegion, updateRegion, sites, navigate})
         updateRegion(newRegion);
     };
 
+    const renderIcons = ({features, facilities}) => {
+        const featureIcons = _.map(features, feature => {
+            return (
+                <Icon
+                    key={feature}
+                    reverse
+                    size={15}
+                    name={featureIconDetails[feature].name}
+                    type={featureIconDetails[feature].type}
+                />
+            )
+        });
+
+        const facilityIcons = _.map(facilities, facility => {
+            return (
+                <Icon
+                    key={facility}
+                    reverse
+                    size={15}
+                    name={facilityIconDetails[facility].name}
+                    type={facilityIconDetails[facility].type}
+                />
+            );
+        });
+
+        return _.concat(featureIcons, facilityIcons);
+    };
+
+    const renderSelectedSitePreview = () => {
+        if (selectedSite && !_.isEmpty(selectedSite)) {
+            const {id, title, description, nearestTown, accessibility, siteImageData, features, facilities} = selectedSite;
+            const {sitePreviewContainerStyle, touchableMainContainerStyle, mainInnerContainerStyle, topRowInfoStyle, topRowText, titleRowStyle, bottomRowInfoStyle, bottomRowText, closeIconStyle, IconContainer} = styles;
+
+            return (
+                <ImageBackground
+                    pointerEvents="none"
+                    source={siteImageData ? {uri: `data:image/png;base64,${siteImageData}`} : require('../../assets/starTent.jpg')}
+                    style={sitePreviewContainerStyle}
+                >
+                    <TouchableOpacity style={touchableMainContainerStyle} onPress={() => getSiteDetail({selectedSite: selectedSite, navigate})}>
```

Comment:
If the user clicks anywhere other than the X button, it means they want to open the `SiteDetailScreen`. Pass the `selectedSite` and `navigate` so we can navigate there.